src/test/java/nl/tudelft/ewi/devhub/server/backend/warnings/LargeCommitWarningGeneratorTest.java

```diff
@@ -0,0 +1,137 @@
+    @Mock private GitPush gitPush;
+    @Mock private List<AbstractDiffModel.DiffFile<AbstractDiffModel.DiffContext<AbstractDiffModel.DiffLine>>> diffs;
+    @Mock private AbstractCommitWarningGenerator abstractCommitWarningGenerator;
+    @Mock private CommitApi commitApi;
+    @Mock private DiffModel diffModel;
+    @Mock private RepositoryApi repositoryApi;
+    @Mock private Collection collection;
+    @Mock private Stream<AbstractDiffModel.DiffFile<AbstractDiffModel.DiffContext<AbstractDiffModel.DiffLine>>> stream;
+    @Mock private Stream<Object> objectStream;
+
+    @Before
+    public void setUp() {
+        generator = new LargeCommitWarningGenerator(repositoriesApi);
+
+        /* Build the mocked commit */
+        when(commit.getRepository()).thenReturn(configurable);
+        when(commit.getRepository().getRepositoryName()).thenReturn(REPOSITORY_NAME);
+        when(commit.getCommitId()).thenReturn(COMMIT_ID);
+        when(commit.isMerge()).thenReturn(false);
+
+        when(abstractCommitWarningGenerator.getRepository(commit)).thenReturn(repositoryApi);
+        when(abstractCommitWarningGenerator.getGitCommit(commit)).thenReturn(commitApi);
+
+        /* Build the mocked repo */
+        when(repositoriesApi.getRepository(REPOSITORY_NAME)).thenReturn(repositoryApi);
+        when(repositoryApi.getCommit(COMMIT_ID)).thenReturn(commitApi);
+        when(commitApi.diff()).thenReturn(diffModel);
+        when(diffModel.getDiffs()).thenReturn(diffs);
+
+        /* Used to prevent NullPointerExceptions in tooManyLineChanges */
+        when(diffs.stream()).thenReturn(stream);
+        when(stream.filter(anyObject())).thenReturn(stream);
+        when(stream.flatMap(anyObject())).thenReturn(objectStream);
+        when(objectStream.flatMap(anyObject())).thenReturn(objectStream);
+        when(objectStream.filter(anyObject())).thenReturn(objectStream);
+
+        /* Set up a warning, used to test if the generator returns an equal object */
+        LargeCommitWarning warning = new LargeCommitWarning();
+        warning.setCommit(commit);
+        testEquals = Sets.newHashSet(warning);
+    }
+
+    /**
+     * Confirms that the warning is not generated when the commit is a merge.
+     */
+    @Test
+    public void testIsMerge() {
+        when(commit.isMerge()).thenReturn(true);
+        Set<LargeCommitWarning> empty = generator.generateWarnings(commit, gitPush);
+        assertTrue(empty.isEmpty());
+    }
+
+    /**
+     * Confirms that the warning is generated when there are too many files committed.
+     */
+    @Test
+    public void testTooManyFiles() {
+        when(diffs.size()).thenReturn(1);
+        when(configurable.getIntegerProperty(anyString(), anyString())).thenReturn(0);
+
+        Set<LargeCommitWarning> warnings = generator.generateWarnings(commit, gitPush);
+
+        verify(diffs).size();
+
+        assertFalse(warnings.isEmpty());
+        assertEquals(1, warnings.size());
```

Comment:
You are basically testing `Collections` here :wink: If warnings has size 1, it can never be empty. The latter assertion makes more sense to me. The first one is redundant :)